### ISSUE TITLE
We need to find way to control random seed and amount of threads at least for tests.

### ISSUE THREAD

**Issue by Ivanidzo4ka** [dotnet/machinelearning#135]
Currently out pipeline.Train function creates
`using (var environment = new TlcEnvironment())`
which is nice, but it doesn't specify seed which become random and concurrency which force algorithms to use all threads available on machine. Which can result in different metrics, predictions and any other stuff we assert in tests.

**Comment by Ivanidzo4ka** [comment]
https://ci2.dot.net/job/Private/job/dotnet_machinelearning/job/master/job/windows_nt_release_prtest/588/testReport/junit/(root)/(empty)/Microsoft_ML_Scenarios_ScenariosTests_TrainAndPredictIrisModelWithStringLabelTest/
as example.
as  https://github.com/dotnet/machinelearning/issues/74 as related issue

**Comment by GalOshri** [comment]
Related to #9

**Comment by TomFinley** [comment]
Hmm, yes. In this case, I suppose, `Rand` and `ConcurrencyFactor`.

Hi @glebuk , hi @KrzysztofCwalina ... here's another issue to add to the pile of those that could be easily addressed just be exposing the existing functionality of `IHostEnvironment`.

**Comment by shauheen** [comment]
closing as some solution was merged and the underlying issue will be handled differently after #371 gets implemented.
